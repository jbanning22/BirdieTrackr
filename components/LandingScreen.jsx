import {
  StyleSheet,
  Text,
  View,
  Image,
  TouchableOpacity,
  Switch,
} from 'react-native';
import React, {useState, useContext} from 'react';
import myImage from '../assets/images/DiscGolfSunset.jpeg';
import {AuthContext} from '../AuthContext';
import AsyncStorage from '@react-native-async-storage/async-storage';

const LandingScreen = ({navigation}) => {
  const [isEnabled, setIsEnabled] = useState(false);
  const {setOffline} = useContext(AuthContext);

  const toggleSwitch = async () => {
    const userData = await AsyncStorage.getItem('userData');
    if (!userData) {
      await AsyncStorage.setItem(
        'userData',
        JSON.stringify({scorecards: [], throws: []}),
      );
    }
    setIsEnabled(previousState => !previousState);
    setOffline(!isEnabled);
  };

  return (
    <View style={styles.box1}>
      <View style={{flexDirection: 'row', alignSelf: 'flex-end', margin: 5}}>
        <Text style={styles.offlineText}>Offline?</Text>
        <Switch
          trackColor={{false: '#2D6061'}}
          thumbColor={isEnabled ? 'white' : '#2D6061'}
          onValueChange={toggleSwitch}
          value={isEnabled}
          testID="offline-switch"
        />
      </View>
      <Image
        source={myImage}
        style={{width: 300, height: 300, borderRadius: 150}}
      />
      <Text style={styles.homeText}>
        Welcome to <Text style={styles.appTitle}>BirdieTrackr</Text>
      </Text>
      <Text style={styles.messageText}>
        Now you can enjoy disc golf with us!
      </Text>
      <TouchableOpacity
        style={styles.loginButton}
        onPress={() => navigation.navigate('SignInPage')}>
        {/* // onPress={() => navigation.navigate('App', {screen: 'Scorecards'})}> */}
        <Text style={styles.textButton}>Sign In with Email</Text>
      </TouchableOpacity>
      <Text style={styles.questionStyle}>Don&apos;t have an account?</Text>
      <TouchableOpacity onPress={() => navigation.navigate('SignUpPage')}>
        <Text style={styles.signUpSize}>Sign Up</Text>
      </TouchableOpacity>
    </View>
  );
};

export default LandingScreen;

const styles = StyleSheet.create({
  box1: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    background: '#7D7D7D',
  },
  loginButton: {
    width: 327,
    height: 56,
    justifyContent: 'center',
    fontFamily: 'Satoshi-Medium',
    alignItems: 'center',
    backgroundColor: '#2D6061',
    borderRadius: 12,
    marginBottom: 40,
    marginTop: 60,
  },
  homeText: {
    fontSize: 28,
    fontWeight: '600',
    marginBottom: 10,
    marginTop: 40,
    fontFamily: 'Satoshi-Medium',
  },
  appTitle: {
    color: '#45B369',
    fontFamily: 'Satoshi-Medium',
  },
  messageText: {
    fontFamily: 'Satoshi-Medium',
    fontSize: 16,
    alignSelf: 'center',
  },
  offlineText: {
    fontFamily: 'Satoshi-Medium',
    fontSize: 16,
    alignSelf: 'center',
    marginRight: 5,
  },
  textButton: {
    fontSize: 18,
    color: 'white',
    fontWeight: '500',
  },
  signUpSize: {
    fontFamily: 'Satoshi-Medium',
    color: '#45B369',
    marginLeft: 5,
    fontSize: 18,
  },
  questionStyle: {
    fontFamily: 'Satoshi-Medium',
    fontSize: 18,
  },
});
