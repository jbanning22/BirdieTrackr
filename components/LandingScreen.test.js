import React from "react";
import {render, screen, fireEvent} from "@testing-library/react-native";
import LandingScreen from "./LandingScreen";

describe("Landing Screen", () => {
	it("should match the snapshot", () => {
		const {toJSON} = render(<LandingScreen />);
		expect(toJSON()).toMatchSnapshot();
	});
	it("should render welcome message", async () => {
		await render(<LandingScreen />);
		expect(screen.getByText("Welcome to IDISC!")).toBeTruthy();
	});
	it("should navigate to sign in screen", async () => {
		const navigation = {
			navigate: jest.fn(),
		};
		const {getByText} = await render(<LandingScreen navigation={navigation} />);
		fireEvent.press(getByText("Sign In"));
		expect(navigation.navigate).toHaveBeenCalledWith("SignInPage");
	});
	it("should navigate to sign up screen", async () => {
		const navigation = {
			navigate: jest.fn(),
		};
		const {getByText} = await render(<LandingScreen navigation={navigation} />);
		fireEvent.press(getByText("Sign Up"));
		expect(navigation.navigate).toHaveBeenCalledWith("SignUpPage");
	});
});
