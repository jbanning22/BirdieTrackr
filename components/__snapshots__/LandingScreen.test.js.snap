// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Screen should match the snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "background": "#7D7D7D",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../assets/images/DiscGolfSunset.jpeg",
      }
    }
    style={
      {
        "borderRadius": 150,
        "height": 300,
        "width": 300,
      }
    }
  />
  <Text
    style={
      {
        "fontFamily": "Satoshi-Medium",
        "fontSize": 28,
        "fontWeight": "600",
        "marginBottom": 10,
        "marginTop": 40,
      }
    }
  >
    Welcome to 
    <Text
      style={
        {
          "color": "#45B369",
          "fontFamily": "Satoshi-Medium",
        }
      }
    >
      DG Scorecard
    </Text>
    !
  </Text>
  <Text
    style={
      {
        "fontFamily": "Satoshi-Medium",
        "fontSize": 16,
      }
    }
  >
    Now you can enjoy disc golf with us!
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#2D6061",
        "borderRadius": 12,
        "fontFamily": "Satoshi-Medium",
        "height": 56,
        "justifyContent": "center",
        "marginBottom": 40,
        "marginTop": 60,
        "opacity": 1,
        "width": 327,
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 18,
          "fontWeight": "500",
        }
      }
    >
      Sign In with Email
    </Text>
  </View>
  <Text
    style={
      {
        "fontFamily": "Satoshi-Medium",
        "fontSize": 18,
      }
    }
  >
    Don't have an account?
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        {
          "color": "#45B369",
          "fontFamily": "Satoshi-Medium",
          "fontSize": 18,
          "marginLeft": 5,
        }
      }
    >
      Sign Up
    </Text>
  </View>
</View>
`;
